packages/client/components/Poll/Poll.tsx

```diff
@@ -0,0 +1,191 @@
+import React, {Ref, useCallback, useMemo} from 'react'
+import styled from '@emotion/styled'
+import {useFragment} from 'react-relay'
+import graphql from 'babel-plugin-relay/macro'
+import {Poll_poll$key} from '~/__generated__/Poll_poll.graphql'
+import {Poll_discussion$key} from '~/__generated__/Poll_discussion.graphql'
+
+import {PollContext, PollState} from './PollContext'
+import {cardShadow, Elevation} from '~/styles/elevation'
+import ThreadedItemWrapper from '../ThreadedItemWrapper'
+import ThreadedAvatarColumn from '../ThreadedAvatarColumn'
+import ThreadedItemHeaderDescription from '../ThreadedItemHeaderDescription'
+import cardRootStyles from '~/styles/helpers/cardRootStyles'
+import {PALETTE} from '~/styles/paletteV3'
+import {updateLocalPollOption, addLocalPollOption, updateLocalPoll} from './local/newPoll'
+import useAtmosphere from '~/hooks/useAtmosphere'
+import CreatePollMutation from '~/mutations/CreatePollMutation'
+import {Polls} from '~/types/constEnums'
+
+const BodyCol = styled('div')({
+  display: 'flex',
+  flexDirection: 'column',
+  paddingBottom: 8,
+  width: '100%',
+  marginTop: 10
+})
+
+const PollRoot = styled('div')<{
+  pollState: PollState
+  isFocused: boolean
+}>(({isFocused, pollState}) => ({
+  ...cardRootStyles,
+  display: 'flex',
+  flexDirection: 'column',
+  justifyContent: 'start',
+  outline: 'none',
+  padding: `0`,
+  overflow: 'hidden',
+  color: PALETTE.SLATE_600,
+  backgroundColor: PALETTE.WHITE,
+  border: `1.5px solid ${
+    isFocused && pollState === 'creating' ? PALETTE.SKY_400 : PALETTE.SLATE_400
+  }`,
+  boxShadow: pollState === 'creating' ? cardShadow : Elevation.Z0,
+  transition: `box-shadow 100ms ease-in`
+}))
+
+interface Props {
+  children: React.ReactNode
+  poll: Poll_poll$key
+  discussion: Poll_discussion$key
+  dataCy: string
+}
+
+const Poll = React.forwardRef((props: Props, ref: Ref<HTMLDivElement>) => {
```

Comment:
Polls are created using compound components pattern, so all the implementation of each part is implemented as separate components and use poll context under the hood. If you see any potential problems with that solution, please let me know!

Reply:
I'm open to changing it. It makes sense to me to design the components that way here. I described that here: https://github.com/ParabolInc/parabol/issues/5145. On the other hand, the 2 downsides you mentioned are valid so if we decide it's too confusing it's not a problem to change it.